x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherPluginTests.java

```diff
@@ -97,4 +103,32 @@ public void testThreadPoolSize() {
                 .build();
         assertThat(Watcher.getWatcherThreadPoolSize(noDataNodeSettings), is(1));
     }
+
+    public void testReload() {
+        Settings settings = Settings.builder()
+            .put("xpack.watcher.enabled", true)
+            .put("path.home", createTempDir())
+            .build();
+        NotificationService mockService = mock(NotificationService.class);
+        Watcher watcher = new Watcher(settings);
+        watcher.reloadableServices.clear();
```

Comment:
instead of exposing the `reloadableServices` via package private visibility, how about making it `protected` and do this (otherwise we have another different visibility in the watcher class)

```
        Watcher watcher = new Watcher(settings) {
            @Override
            public Collection<Object> createComponents(Client client, ClusterService clusterService, ThreadPool threadPool,
                                                       ResourceWatcherService resourceWatcherService, ScriptService scriptService,
                                                       NamedXContentRegistry xContentRegistry, Environment environment,
                                                       NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry) {
                reloadableServices.add(mockService);
                return Collections.emptyList();
            }
        };

```

Reply:
👍